MbsAPI/f_evt.c

```diff
@@ -2471,107 +2714,136 @@ INTS4 f_evt_cre_tagfile(CHARS* pc_lmd, CHARS* pc_tag,INTS4 (*e_filter)())
+    }
+    if (NULL == ps_chan->ps_taghe) {
+        close(ps_chan->l_tagfile_no);
+        return (GETEVT__TAGRDERR);
+    }
+    if (ps_chan->ps_taghe->l_endian != 1) {
+        ps_chan->l_tagswap = 1;
+    }
+    if (ps_chan->l_tagswap) {
+        f_evt_swap((CHARS*)ps_chan->ps_taghe, sizeof(s_taghe));
+    }
+    if (l_prihe) {
+        printf("Buffers %6d, of size %d, Events %10d, first %d, last %d",
+               ps_chan->ps_taghe->l_buffers,
+               ps_chan->ps_taghe->l_bufsize,
+               ps_chan->ps_taghe->l_events,
+               ps_chan->ps_taghe->l_first,
+               ps_chan->ps_taghe->l_last);
+        if (ps_chan->ps_taghe->l_linear) {
+            printf(" linear\n");
+        } else {
+            printf(" not linear\n");
+        }
+    }
+    if ((ps_chan->l_channel_no = open(pc_lmd, GET__OPEN_FLAG)) == -1) {
+        if (ps_chan->ps_taghe != NULL) {
+            free(ps_chan->ps_taghe);
+        }
+        ps_chan->ps_taghe = NULL;
+        close(ps_chan->l_tagfile_no);
+        return (GETEVT__NOFILE);
+    }
+    /* read buffer header to check if we have to swap */
+    ps_bufhe = (s_bufhe*)c_temp;
+    if (ps_chan->ps_taghe->l_bufsize >= 0 && ps_chan->ps_taghe->l_bufsize < sizeof(c_temp)) {
+        if (read(ps_chan->l_channel_no, c_temp, ps_chan->ps_taghe->l_bufsize) != ps_chan->ps_taghe->l_bufsize) {
+            if (ps_chan->ps_taghe != NULL) {
+                free(ps_chan->ps_taghe);
+            }
+            ps_chan->ps_taghe = NULL;
+            if (ps_chan->ps_tag != NULL) {
+                free(ps_chan->ps_tag);
+            }
+            ps_chan->ps_tag = NULL;
+            close(ps_chan->l_tagfile_no);
+            close(ps_chan->l_channel_no);
+            return (GETEVT__RDERR);
+        }
+    }
+    if (ps_chan->ps_taghe != NULL) {
+        if (ps_chan->ps_taghe->l_linear == 0) {
+            if (ps_chan->ps_taghe->l_filesize >= 0 && ps_chan->ps_taghe->l_filesize < sizeof(s_tag)) {
+                ps_chan->ps_tag = (s_tag*)malloc(ps_chan->ps_taghe->l_filesize);
+                if (read(ps_chan->l_tagfile_no, (CHARS*)ps_chan->ps_tag, ps_chan->ps_taghe->l_filesize)
+                    != ps_chan->ps_taghe->l_filesize) {
+                    if (ps_chan->ps_taghe != NULL) {
+                        free(ps_chan->ps_taghe);
+                    }
+                    ps_chan->ps_taghe = NULL;
+                    if (ps_chan->ps_tag != NULL) {
+                        free(ps_chan->ps_tag);
+                    }
+                    ps_chan->ps_tag = NULL;
+                    close(ps_chan->l_tagfile_no);
+                    close(ps_chan->l_channel_no);
+                    return (GETEVT__TAGRDERR);
+                }
+            }
+        }
+    }
+    if (ps_bufhe->l_free[0] != 1) {
+        ps_chan->l_lmdswap = 1;
+    }
+    if (ps_chan->l_lmdswap) {
+        f_evt_swap_filhe(ps_bufhe);
+    }
+    if (l_prihe) {
+        ii = f_evt_type(ps_bufhe, NULL, 0, 0, 0, 0);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'ii' is assigned a value that is never used.](https://app.codacy.com/manual/FairRootGroup/FairRoot/pullRequest?prid=5323596)